mda_openbabel_converter/tests/test_openbabel_parser.py

```diff
@@ -0,0 +1,132 @@
+
+    def test_mandatory_attributes(self, top):
+        for attr in self.mandatory_attrs:
+            assert hasattr(top, attr), 'Missing required attribute: {}'.format(attr)
+
+    def test_attrs_total_counts(self, top):
+        ag = mda.Universe(top).select_atoms("all")
+        res = ag.residues
+        seg = ag.segments
+        assert len(ag) == self.expected_n_atoms
+        assert len(res) == self.expected_n_residues
+        assert len(seg) == self.expected_n_segments
+
+
+class TestOpenBabelParserSMILES(OpenBabelParserBase):
+    #parser = mda_openbabel_converter.OpenBabelParser.OpenBabelParser
+    expected_attrs = OpenBabelParserBase.expected_attrs + ['charges']
+
+    @pytest.fixture()
+    def filename(self):
+        obConversion = ob.OBConversion()
+        obConversion.SetInFormat("smi")
+        mol = OBMol()
+        obConversion.ReadString(mol, "C1=CC=CS1")
+        mol.AddHydrogens()
+        return mol
+    
+    @pytest.fixture()
+    def top(self, filename):
+        yield self.parser(filename).parse()
+        
+    expected_n_atoms = 9
+    expected_n_residues = 1
+    expected_n_segments = 1
+    expected_n_bonds = 9
+
+    def test_attrs_total_counts(self, top):
+        u = mda.Universe(top)
+        ag = u.select_atoms("all")
+        res = ag.residues
+        seg = ag.segments
+        assert len(ag) == self.expected_n_atoms
+        assert len(res) == self.expected_n_residues
+        assert len(seg) == self.expected_n_segments
+
+    def test_aromaticities(self, top, filename):
+        expected = np.array([
+            atom.IsAromatic() for atom in ob.OBMolAtomIter(filename)])
+        assert_equal(expected, top.aromaticities.values)
+
+    def test_elements(self, top, filename):
+        expected = np.array([
+            OBElementTable().GetSymbol(atom.GetAtomicNum()) for atom in ob.OBMolAtomIter(filename)])
+        assert_equal(expected, top.elements.values)
+
+    # def test_chiralities(self, top, filename):
+    #     chirality = None
+    #     if atom.IsPositiveStereo():
+    #         chirality = "+"
+    #     if atom.IsNegativeStereo():
+    #         chirality = "-"
+    #     chiralities.append(chirality)
+
+    #     expected = np.array([
+    #         OBElementTable().GetSymbol(atom.GetAtomicNum()) for atom in ob.OBMolAtomIter(filename)])
+    #     assert_equal(expected, top.chiralities.values)
+
+    def test_charges(self, top, filename):
+        expected = np.array([
+            atom.GetPartialCharge() for atom in ob.OBMolAtomIter(filename)])
+        assert_allclose(expected, top.charges.values)
+
+    # def test_mass_check(self, top, filename):
+    #     expected = np.array([
+    #         OBElementTable().GetSymbol(atom.GetAtomicNum()) for atom in ob.OBMolAtomIter(filename)])
+    #     assert_equal(expected, top.elements.values)
+
+    # need to check other attrs including:
+    # 'ids', 'names', 'resids', 'resnums', 'chiralities', 'segids', 'charges'
```

Comment:
I would add at least three more OBmols originating from files to test. Try and vary the chemistry, something large and protein-like perhaps, an SDF with a complex organic ligand, and  something rather metallic.  You can set these up in `conftest.py`